Sources/Workspace/Workspace.swift

```diff
@@ -1544,17 +1548,27 @@ extension Workspace {
         for (packageRef, manifest) in packageAndManifests {
             for target in manifest.targets where target.type == .binary {
                 if let path = target.path {
-                    let artifact = ManagedArtifact.local(packageRef: packageRef, targetName: target.name, path: path)
-                    artifacts.append(artifact)
+                    // 👀 FIXME: is there a better way?
+                    // the target path is validated earlier to be within the package directory
+                    let absolutePath = manifest.path.parentDirectory.appending(component: path)
+                    artifacts.append(
+                        .local(
+                            packageRef: packageRef,
+                            targetName: target.name,
+                            path: absolutePath)
+                    )
                 } else if let url = target.url, let checksum = target.checksum {
-                    let subpath = artifactSubpath(packageRef: packageRef, targetName: target.name)
-                    let artifact = ManagedArtifact.remote(
-                        packageRef: packageRef,
-                        targetName: target.name,
-                        url: url,
-                        checksum: checksum,
-                        subpath: subpath)
-                    artifacts.append(artifact)
+                    // 👀 FIXME: hard coding "xcframework" here is obviously wrong, but need to understand what we expect today in the
+                    // zipfile to see how we can fix this
+                    let subpath = RelativePath("\(packageRef.name)/\(target.name).xcframework")
```

Comment:
this is the one missing piece to make this PR actually useful. as is stands, the code always downloads into a "xcframework" directory. I need to better understand the structure of the binary dependencies zipfile. if it has a top level "xcframework" directory, I could fairly easily make it work. if it is "flat" and has the xcframework content at the root we would need to come up with something else which may require configuration :(

Reply:
My understanding is that the .zip should contain a .xcframework at the top level.  It's seems pretty clear from the various instructions for creating them that the zip is created with `zip -r foo.xcframework.zip foo.xcframework`, so the .xcframework ends up as the sole entry at the top of the zip file.  Seems to me that we could probably require the .zip to contain a single entry, and then use the suffix of that entry to determine what it is.